lib/shim/webframework-shim.js

```diff
@@ -677,6 +677,23 @@ function _defaultResponsePredicate() {
   return false
 }
 
+/**
+ * Attaches a flag on function indicating that
+ * CLM attributes need to be associated with the middleware
+ * span during record
+ *
+ * Note: director middleware instrumentation passes in a string as the "function" so we have to check if the middleware is actually a function
+ * to avoid crashing.
+ *
+ * @param {Shim} shim instance of shim
+ * @param {Function} middleware function to apply clm symbol
+ */
+function assignCLMSymbol(shim, middleware) {
```

Comment:
i think we can eventually just remove this and move it all to the code-level-metrics util then we'd be decorating every function that uses record

Reply:
Just to confirm my understanding, we can't do that right now because we have CLM behind a config flag, correct?

Reply:
no, we need to do some OATS testing and napkin math on the size increase of segments/spans.  If we move the checking of config to the `lib/util/code-level-metrics.addCLMAttributes` it'd attach CLM attrs to almost all spans for functions we instrument. That's because most instrumentation flows through record which is where `addCLMAttributes` is getting called atm. Since most of the spans won't be in user code it'd be pointless

Reply:
I was confused why we'd ever want to annotate all spans if most wouldn't need the CLM attributes, but like you said, we'd only move it if we ever made enhancements to the point where having CLM attributes on everything made sense, got it.

Reply:
nothing is perfect because right now this will hit middleware core to a framework or in a library. you can see that in the express versioned tests the body parser and express init middleware have CLM attrs